### ISSUE TITLE
This is awesome

### ISSUE THREAD

**Issue by arthurwolf** [tapnair/UGS_Fusion#1]
I just wanted to say this is really really great ! I posted about it to the Smoothieware community.

I actually wanted to write a plugin that allows users to send files and control their Smoothieboard from within Fusion360. Is that something you'd be able to help a tiny bit with if I run into problems I can't solve myself ? It's something a lot of people would love, so I'll take all the help I can get to make it happen.

**Comment by tapnair** [comment]
So I have thought about trying to do something like this, but i really think that just the export and hand off is the right move.  The Fusion API is running inside the main process and if you tried to run smoothie from within Fusion it would be weird I think.  Better to start it in its own thread and send the relevant commands over there.

**Comment by tapnair** [comment]
Now all that being said you could set up the ability to do basic things like homing, etc from within a Fusion add-in. If you check out my Octoprint one i do some of that by just making calls to the local server API.  Maybe it would work?  But i am not so sure.  Of course i would be super happy to help in any way i can just to see what is possible.

**Comment by arthurwolf** [comment]
Yes, the goal would be to allow the user to do basic "setup" tasks such as z-probing, edge finding, setting up the origin etc. Those are very simple tasks ( fusion-side ) : we just need to send gcodes over HTTP.

I have a question : if I understand correctly, fusion is going to be browser-based at some point. I expect that means any plugin coded in Python won't work at that point. Would coding plugins in javascript ensure they'll still work ?

**Comment by tapnair** [comment]
The browser based version o Fusion is still a ways out, especially for CAM.  So for now i wouldn't worry too much about it.  Sorry for the delay on the response.  Let me know how it goes.  It would be cool to put this is the Fusion app store if you are successful.

**Comment by Richardrodriguez123** [comment]
Hey thanks for the add-on.
I am trying to use it and when i post to ugs everything seems fine and it starts to open ugs platform and then fails to open. Have you encountered this or can walk me through the possible solution?
Thanks in advance!!
